mantidproject/mantid
# Surface and contour plots versus log values from workspace groups

Resolves #14739 

On right-clicking on a workspace group, the user is offered the option to plot a 3D surface or contour plot (as long as the group contains more than 2 workspaces). The option is shown as long as the workspace clicked on can be cast to a const WorkspaceGroup and contains >2 workspaces.

A dialog enables the choice of which spectrum to use from each workspace, and also what the Y axis of the plot will be. (X is the x-axis of the workspaces, Z is the data value). The default option is workspace index, and the other options are all the sample logs with single, numeric values per workspace. There is also a "custom" option which allows the user to type their values in as a comma-separated list (as an alternative to using `AddSampleLog`).

The [release notes](http://www.mantidproject.org/ReleaseNotes_3_6_UI_Changes) have been updated.

**Test:**
Use the script below to generate an example WorkspaceGroup of 10 workspaces. 
- Check that right-clicking on the group gives options to plot a surface or contour plot
- Check these options are not shown when clicking on a group of 2 or fewer workspaces, or a group that contains table/peaks workspaces
- Generate plots of each type, using the dialog to choose the Y axis - try the `MyLog` log and also using the custom input box (enter a comma-separated list of 10 values)
- Check layout of dialog is ok
- Code review: check no excessive use of ADS
- Code review: verify not too much bloat has been added to MantidDockWidget and instead the new code is refactored into its own class

**Example data to test:**
The script below creates a minimal data set, and the plots can also be compared to [Pascal Manuel's scripts](https://gist.github.com/tom-perkins/998caf72095c54258571) if you have access to the Dropbox data.

``` python
def create_sample_group(n_items):
    to_group = list()
    for i in range(n_items):
        name = 'ws' + str(i)
        to_group.append(name)
        CreateWorkspace(DataX=range(200, 210, 1), DataY=[i+1]*20, NSpec=2, OutputWorkspace=name)
        AddSampleLog(Workspace=name, LogName="MyLog", LogText=str((i+1)*5), LogType='Number')
    group = GroupWorkspaces(InputWorkspaces=','.join(to_group))
    return group

create_sample_group(n_items=10)
```
